### ISSUE TITLE
Build fails (hyprpm)

### ISSUE THREAD

**Issue by allylikesu** [Duckonaut/split-monitor-workspaces#56]
I'm quite new to the Hyprland ecosystem, apologies as I'm not very familiar with it.

I have the latest Hyprland installed (self-compiled, commit 220144276b8789d05e58df3e0452314c11b64728)

Output from running `hyprpm -v add https://github.com/Duckonaut/split-monitor-workspaces`:
```
allison ~ % hyprpm -v add https://github.com/Duckonaut/split-monitor-workspaces                                                                                                                                                       [ OK ]
[v] version returned: Hyprland, built from branch main at commit 220144276b8789d05e58df3e0452314c11b64728  (layout: unfullscreen on toggle into tiled fullscreen).
Date: Sun Mar 10 17:31:49 2024
Tag: v0.36.0-97-g22014427

flags: (if any)


[v] parsed commit 220144276b8789d05e58df3e0452314c11b64728 at branch main
✔ adding a new plugin repository from https://github.com/Duckonaut/split-monitor-workspaces
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n] 

 → Cloning https://github.com/Duckonaut/split-monitor-workspaces                                                                                                                                                                             
✔ cloned                                                                                                                                                                                                                                     
✔ found hyprpm manifest                                                                                                                                                                                                                      
✔ parsed manifest, found 1 plugins:                                                                                                                                                                                                          
 → split-monitor-workspaces by Duckonaut version                                                                                                                                                                                             
✔ Hyprland headers OK                                                                                                                                                                                                                        
 → Building split-monitor-workspaces                                                                                                                                                                                                         
✖ Plugin split-monitor-workspaces failed to build.                                                                                                                                                                                           

 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━  3 / 5  Building plugin(s)[v] shell returned:  -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/allison/.local/share/hyprpm/headersRoot/share/pkgconfig" meson setup build --wipe         
The Meson build system
Version: 1.3.2
Source dir: /tmp/hyprpm/new
Build dir: /tmp/hyprpm/new/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: ccache c++ (gcc 13.2.1 "c++ (GCC) 13.2.1 20230801")
C++ linker for the host machine: c++ ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES 
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.0
Run-time dependency hyprland found: YES 0.36.0
Run-time dependency pixman-1 found: YES 0.43.4
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

Found ninja-1.11.1 at /usr/bin/ninja
                                                
 -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/allison/.local/share/hyprpm/headersRoot/share/pkgconfig" meson compile -C build
ninja: Entering directory `/tmp/hyprpm/new/build'
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o 
ccache c++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/protocols -I/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/wlroots -I/home/allison/.local/share/hyprpm/headersRoot/include -I/usr/include/pixman-1 -I/usr/include/libdrm -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
In file included from /home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../Window.hpp:8,
                 from /home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/config/ConfigManager.hpp:16,
                 from /home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:10,
                 from ../src/main.cpp:3:
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../config/ConfigDataValues.hpp: In member function ‘virtual std::string CGradientValueData::toString()’:
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../config/ConfigDataValues.hpp:64:5: warning: no return statement in function returning non-void [-Wreturn-type]
   64 |     }
      |     ^
In file included from ../src/main.cpp:6:
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp: At global scope:
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:7:6: error: multiple definition of ‘enum eFullscreenMode’
    7 | enum eFullscreenMode : int8_t {
      |      ^~~~~~~~~~~~~~~
In file included from /home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:24:
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:7:6: note: previous definition here
    7 | enum eFullscreenMode : int8_t {
      |      ^~~~~~~~~~~~~~~
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:15:7: error: redefinition of ‘class CWorkspace’
   15 | class CWorkspace {
      |       ^~~~~~~~~~
/home/allison/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:15:7: note: previous definition of ‘class CWorkspace’
   15 | class CWorkspace {
      |       ^~~~~~~~~~
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/hyprpm/new/build


✔ all plugins built                                                                                                                                                                                                                          
✔ installed repository                                                                                                                                                                                                                       
✔ you can now enable the plugin(s) with hyprpm enable                                                                                                                                                                                        
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!
```

**Comment by lukeemhigh** [comment]
I can reproduce

**Comment by Expressive-Synapse** [comment]
Im struggling with the exact same issue. Im on arch using hyprland 0.37.1-1

**Comment by avih7531** [comment]
Same issue here. Just updated to hyprland 0.37.1-1 on Arch:
```
╰─❯ hyprpm -v add https://github.com/Duckonaut/split-monitor-workspaces
[v] version returned: Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)


[v] parsed commit c5e28ebcfe00a510922779b2c568cfa52a317445 at branch HEAD
✔ adding a new plugin repository from https://github.com/Duckonaut/split-monitor-workspaces
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n]

 → Cloning https://github.com/Duckonaut/split-monitor-workspaces
✔ cloned
✔ found hyprpm manifest
✔ parsed manifest, found 1 plugins:
 → split-monitor-workspaces by Duckonaut version
✔ Hyprland headers OK
 → Building split-monitor-workspaces
✖ Plugin split-monitor-workspaces failed to build.

 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━  3 / 5  Building plugin(s)[v] shell returned:  -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/aviherman/.local/share/hyprpm/headersRoot/share/pkgconfig" meson setup build --wipe
The Meson build system
Version: 1.4.0
Source dir: /tmp/hyprpm/new
Build dir: /tmp/hyprpm/new/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: c++ (gcc 13.2.1 "c++ (GCC) 13.2.1 20230801")
C++ linker for the host machine: c++ ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency hyprland found: YES 0.37.1
Run-time dependency pixman-1 found: YES 0.43.4
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

Found ninja-1.11.1 at /usr/bin/ninja

 -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/aviherman/.local/share/hyprpm/headersRoot/share/pkgconfig" meson compile -C build
ninja: Entering directory `/tmp/hyprpm/new/build'
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
c++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/protocols -I/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/wlroots -I/home/aviherman/.local/share/hyprpm/headersRoot/include -I/usr/include/pixman-1 -I/usr/include/libdrm -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
In file included from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../Window.hpp:8,
                 from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/config/ConfigManager.hpp:16,
                 from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:10,
                 from ../src/main.cpp:3:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../config/ConfigDataValues.hpp: In member function ‘virtual std::string CGradientValueData::toString()’:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/config/../config/ConfigDataValues.hpp:64:5: warning: no return statement in function returning non-void [-Wreturn-type]
   64 |     }
      |     ^
In file included from ../src/main.cpp:6:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp: At global scope:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:7:6: error: multiple definition of ‘enum eFullscreenMode’
    7 | enum eFullscreenMode : int8_t {
      |      ^~~~~~~~~~~~~~~
In file included from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:24:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:7:6: note: previous definition here
    7 | enum eFullscreenMode : int8_t {
      |      ^~~~~~~~~~~~~~~
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:15:7: error: redefinition of ‘class CWorkspace’
   15 | class CWorkspace {
      |       ^~~~~~~~~~
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:15:7: note: previous definition of ‘class CWorkspace’
   15 | class CWorkspace {
      |       ^~~~~~~~~~
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/hyprpm/new/build


✔ all plugins built
✔ installed repository
✔ you can now enable the plugin(s) with hyprpm enable
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!
```

**Comment by chriselrod** [comment]
Huh
```
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp: At global scope:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:7:6: error: multiple definition of ‘enum eFullscreenMode’
    7 | enum eFullscreenMode : int8_t {
      |      ^~~~~~~~~~~~~~~
In file included from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:24:
/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:7:6: note: previous definition here
```
Multiple copies of the same file defeats `#pragma once`, resulting in multiple definition errors.

But, why are there multiple copies of the same file?!?!?

https://github.com/hyprwm/Hyprland/blob/1c460e98f870676b15871fe4e5bfeb1a32a3d6d8/src/helpers/Workspace.hpp#L7
Note, this file exists in 0.36, but NOT in 0.37!
https://github.com/hyprwm/Hyprland/blob/19c90048d65a5660384d2fb865926a366696d6be/src/desktop/Workspace.hpp#L7
This file does NOT exist in 0.36, but does in 0.37!

Seems like, rather than cloning the repo and checking out the latest, `hyprpm` is somehow merging everything.

So, I haven't tried this yet, but I suggest wiping
`~/.local/share/hyprpm/`
and seeing if that fixes the build, by giving us one clean copy of the latest hyprland headers.

**Comment by avih7531** [comment]
Hi, thanks for your response.

I did a successful `sudo rm -rf ~/.local/share/hyprpm/` and then a `hyprpm add https://github.com/Duckonaut/split-monitor-workspaces/` which led me to 

```
[v] version returned: Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)


[v] parsed commit c5e28ebcfe00a510922779b2c568cfa52a317445 at branch HEAD
! Disclaimer:
  plugins, especially not official, have no guarantee of stability, availablity or security.
  Run them at your own risk.
  This message will not appear again.
✔ adding a new plugin repository from https://github.com/Duckonaut/split-monitor-workspaces/
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n]

 → Cloning https://github.com/Duckonaut/split-monitor-workspaces/
✔ cloned
✔ found hyprpm manifest
✔ parsed manifest, found 1 plugins:
 → split-monitor-workspaces by Duckonaut version
 ━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  2 / 5  Verifying headers
✖ Headers missing. Please run hyprpm update to fix those.
```
So I went and ran `hyprpm update` which told me it "installed headers".

After that I ran `https://github.com/Duckonaut/split-monitor-workspaces/` again which led to me to this error:

```
[v] version returned: Hyprland, built from branch HEAD at commit c5e28ebcfe00a510922779b2c568cfa52a317445  (props: bump ver 0.37.1).
Date: Sat Mar 16 14:51:49 2024
Tag: v0.37.1

flags: (if any)


[v] parsed commit c5e28ebcfe00a510922779b2c568cfa52a317445 at branch HEAD
✔ adding a new plugin repository from https://github.com/Duckonaut/split-monitor-workspaces/
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n]

! old plugin repo build files found in temp directory, removing.
 → Cloning https://github.com/Duckonaut/split-monitor-workspaces/
✔ cloned
✔ found hyprpm manifest
✔ parsed manifest, found 1 plugins:
 → split-monitor-workspaces by Duckonaut version
✔ Hyprland headers OK
 → Building split-monitor-workspaces
✖ Plugin split-monitor-workspaces failed to build.

 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━  3 / 5  Building plugin(s)[v] shell returned:  -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/aviherman/.local/share/hyprpm/headersRoot/share/pkgconfig" meson setup build --wipe
The Meson build system
Version: 1.4.0
Source dir: /tmp/hyprpm/new
Build dir: /tmp/hyprpm/new/build
Build type: native build
Project name: split-monitor-workspaces
Project version: 0.1
C++ compiler for the host machine: c++ (gcc 13.2.1 "c++ (GCC) 13.2.1 20230801")
C++ linker for the host machine: c++ ld.bfd 2.42.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C++ supports arguments -std=c++23: YES
meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
Found pkg-config: YES (/usr/bin/pkg-config) 2.1.1
Run-time dependency hyprland found: YES 0.37.1
Run-time dependency pixman-1 found: YES 0.43.4
Run-time dependency libdrm found: YES 2.4.120
Build targets in project: 1

Found ninja-1.11.1 at /usr/bin/ninja

 -> cd /tmp/hyprpm/new && PKG_CONFIG_PATH="/home/aviherman/.local/share/hyprpm/headersRoot/share/pkgconfig" meson compile -C build
ninja: Entering directory `/tmp/hyprpm/new/build'
[1/2] Compiling C++ object libsplit-monitor-workspaces.so.p/src_main.cpp.o
FAILED: libsplit-monitor-workspaces.so.p/src_main.cpp.o
c++ -Ilibsplit-monitor-workspaces.so.p -I. -I.. -I../include -I/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/protocols -I/home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/wlroots -I/home/aviherman/.local/share/hyprpm/headersRoot/include -I/usr/include/pixman-1 -I/usr/include/libdrm -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O3 -std=c++23 -DWLR_USE_UNSTABLE -fPIC -MD -MQ libsplit-monitor-workspaces.so.p/src_main.cpp.o -MF libsplit-monitor-workspaces.so.p/src_main.cpp.o.d -o libsplit-monitor-workspaces.so.p/src_main.cpp.o -c ../src/main.cpp
../src/main.cpp:6:10: fatal error: hyprland/src/helpers/Workspace.hpp: No such file or directory
    6 | #include <hyprland/src/helpers/Workspace.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /tmp/hyprpm/new/build


✔ all plugins built
✔ installed repository
✔ you can now enable the plugin(s) with hyprpm enable
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!
```

**Comment by Expressive-Synapse** [comment]
> Huh
> 
> ```
> /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp: At global scope:
> /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/helpers/Workspace.hpp:7:6: error: multiple definition of ‘enum eFullscreenMode’
>     7 | enum eFullscreenMode : int8_t {
>       |      ^~~~~~~~~~~~~~~
> In file included from /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/Compositor.hpp:24:
> /home/aviherman/.local/share/hyprpm/headersRoot/include/hyprland/src/desktop/Workspace.hpp:7:6: note: previous definition here
> ```
> 
> Multiple copies of the same file defeats `#pragma once`, resulting in multiple definition errors.
> 
> But, why are there multiple copies of the same file?!?!?
> 
> https://github.com/hyprwm/Hyprland/blob/1c460e98f870676b15871fe4e5bfeb1a32a3d6d8/src/helpers/Workspace.hpp#L7 Note, this file exists in 0.36, but NOT in 0.37! https://github.com/hyprwm/Hyprland/blob/19c90048d65a5660384d2fb865926a366696d6be/src/desktop/Workspace.hpp#L7 This file does NOT exist in 0.36, but does in 0.37!
> 
> Seems like, rather than cloning the repo and checking out the latest, `hyprpm` is somehow merging everything.
> 
> So, I haven't tried this yet, but I suggest wiping
> 
> ```shell
> rm -rf ~/.local/share/hyprpm
> ```
> 
> and seeing if that fixes the build, by giving us one clean copy of the latest hyprland headers, instead of some monstrous union of all the cumulative releases.

It did not seem to resolve the build issue. I have however found that for some reason hyprpm cannot remove /tmp/hyprpm files which requires manual intervention.

**Comment by avih7531** [comment]
Got it -- any ideas for how to proceed from my most recent update? Thank you!

**Comment by chriselrod** [comment]
@avih7531 Ah, of course, obviously hyprland isn't going to be including the non-existant file, but the outdated split-monitor-workspaces might. #60 fixes it for me.
```sh
> hyprpm add https://github.com/chriselro
d/split-monitor-workspaces 9c5486e2014de6929c21adbaf1578fc630530971
✔ adding a new plugin repository from https://github.com/chriselrod/split-monitor-workspaces
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n] 

 ╍━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  0 / 5  Cloning the plugin reposito → Cloning https://github.com/chriselrod/split-monitor-workspaces                      
 ╍━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  0 / 5  Cloning the plugin reposito✔ cloned                                                                               
 ━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  1 / 5  Cloning the plugin reposito✔ found hyprpm manifest                                                                
✔ parsed manifest, found 1 plugins:                                                    
 → split-monitor-workspaces by Duckonaut version                                       
✔ Hyprland headers OK                                                                  
 → Building split-monitor-workspaces                                                   
✔ built split-monitor-workspaces into build/libsplit-monitor-workspaces.so             
✔ all plugins built                                                                    
✔ installed repository                                                                 
✔ you can now enable the plugin(s) with hyprpm enable                                  
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!                      
```

**Comment by Expressive-Synapse** [comment]
now I just need to learn how to use the fix

**Comment by avih7531** [comment]
You fixed it, thanks! I'm a git newb, so possible dumb question, but given that `9c5486e2014de6929c21adbaf1578fc630530971` number at the end of the `git clone` fixed the issue, will that eventually merge into the normal github so I won't have to clone it like that?

Also, I have a few other plugins that are not Hyprland 3.7.1-1 ready (HyprFocus especially). If it was a quick and easy fix, what did you do so I can submit a pull request?

Thank you so much!!

**Comment by Expressive-Synapse** [comment]
> You fixed it, thanks! I'm a git newb, so possible dumb question, but given that `9c5486e2014de6929c21adbaf1578fc630530971` number at the end of the `git clone` fixed the issue, will that eventually merge into the normal github so I won't have to clone it like that?
> 
> Also, I have a few other plugins that are not Hyprland 3.7.1-1 ready (HyprFocus especially). If it was a quick and easy fix, what did you do so I can submit a pull request?
> 
> Thank you so much!!

Okay so I can answer that first question. The fix will get merged with the main branch only if the person in charge of the branch or someone with permissions to do so manually merges the fix, this should happen eventually though assuming the fix doesn't have any problems, and considering its just the switch up of 2 lines of code, it could be merged without issue so long as someone in charge notices the request. Of course im also a bit new to git so take what I say with a grain of salt.

**Comment by chriselrod** [comment]
> now I just need to learn how to use the fix

I did

```sh
hyprpm remove split-monitor-workspaces
hyprpm add https://github.com/chriselro
d/split-monitor-workspaces 9c5486e2014de6929c21adbaf1578fc630530971
```
This removes the old one, and then installs the new one using commit 9c5486 from my fork.

However, you probably don't want to do that long term, since that is a specific commit (i.e., it's immutable/won't see any changes).

> You fixed it, thanks! I'm a git newb, so possible dumb question, but given that 9c5486e2014de6929c21adbaf1578fc630530971 number at the end of the git clone fixed the issue, will that eventually merge into the normal github so I won't have to clone it like that?

I opened PR #60 against this repository.
When someone with merge rights merges it, it will be a part of this repository.

> If it was a quick and easy fix, what did you do so I can submit a pull request?

You can see what I did by clicking the PR, and then clicking "files changed". There, you can see that the only change was that I updated the path: `Workspace.hpp` moved from `src/helpers` to `src/desktop`. I also kept the includes in alphabetical order, since that's a common style.
https://github.com/Duckonaut/split-monitor-workspaces/pull/60/files

**Comment by avih7531** [comment]
Super cool -- appreciate the explanation! 

Oh benevolent dictator of split-monitor-workspace, please merge this!

**Comment by chriselrod** [comment]
> Okay so I can answer that first question. The fix will get merged with the main branch only if the person in charge of the branch or someone with permissions to do so manually merges the fix, this should happen eventually though assuming the fix doesn't have any problems, and considering its just the switch up of 2 lines of code, it could be merged without issue so long as someone in charge notices the request. Of course im also a bit new to git so take what I say with a grain of salt.

Yes. If you're watching a repo (most people watch at least the repos they own), you'll get a notification whenever someone opens a PR.
Such a trivial change is easy to review.

The primary objection my PR is likely to get is that it breaks compatibility with hyprland 0.36.
That is, someone on 0.36 will get basically the same error with my commit as we do on the main branch of split-monitor-workspaces with hyprland 0.37.

The author might demand things like also updating https://github.com/Duckonaut/split-monitor-workspaces/blob/d0012b8b0f764e32dd7b82f7a94b8c30197d7dc8/flake.lock#L16 and the rest of the dependencies to reflect these changes.
Maybe there is some automated way to do that? I'm not familiar with Nix.

But having some dependency management here would be nice. Seems sorely missing among Hyprland plugins.

**Comment by avih7531** [comment]
Thank you!

**Comment by tharrmeehan** [comment]
if this does not get pushed, just use this plugin:
https://github.com/shezdy/hyprsplit

it atleast works.

**Comment by lukeemhigh** [comment]
Is this project still maintained? Not trying to be polemical, I'm genuinely asking

**Comment by Duckonaut** [comment]
Fixed in #52
